File: python/bullmq/types/queue_events_options.py
from typing import TypedDict, Any, Union
import redis.asyncio as redis


class QueueEventsOptions(TypedDict, total=False):
    """
    Options for the QueueEvents class.
    """

    prefix: str
    """
    Prefix for all queue keys.
    """

    backend: str
    """
    Datastore backend to use: 'redis' (default) or 'postgres'. Must match the
    backend the queue itself was created with.

    @default 'redis'
    """

    connection: Union[dict[str, Any], redis.Redis, str]
    """
    Options for connecting to the datastore (a libpq connection string or
    parameter dict when `backend` is 'postgres'). QueueEvents requires a
    dedicated connection because the blocking stream read ties up the
    connection for the duration of the call; passing a shared connection
    would starve other operations.
    """

    schema: str
    """
    PostgreSQL schema used to namespace queues (the SQL-native replacement for
    `prefix`). Only used when `backend` is 'postgres'.

    @default 'bullmq'
    """

    autorun: bool
    """
    Whether to immediately start consuming events on construction.

    @default True
    """

    blockingTimeout: int
    """
    Block timeout in milliseconds for the underlying XREAD BLOCK call.
    A larger value reduces wakeups but increases shutdown latency
    because the consumer loop only checks the closing flag between
    blocks.

    @default 10000
    """

    lastEventId: str
    """
    Stream id to resume from. Use a concrete id (e.g. the last id
    seen by a previous consumer) to replay events, or '$' to only
    receive events published after the consumer attached.

    @default '$'
    """

    skipVersionCheck: bool
    """
    Avoid version validation to be greater or equal than v5.0.0.

    @default False
    """


class QueueEventsProducerOptions(TypedDict, total=False):
    """
    Options for the QueueEventsProducer class. Producers do not need
    a dedicated connection because XADD is non-blocking.
    """

    prefix: str
    backend: str
    connection: Union[dict[str, Any], redis.Redis, str]
    schema: str
    skipVersionCheck: bool

File: java/src/main/java/io/bullmq/core/options/QueueEventsOptions.java
package io.bullmq.core.options;

/**
 * Options for a {@link io.bullmq.core.events.QueueEvents QueueEvents} listener.
 */
public class QueueEventsOptions extends QueueBaseOptions {
    /**
     * When true (default) the listener starts consuming events immediately.
     */
    private boolean autorun = true;

    /**
     * Cursor to start from. Defaults to <c>"$"</c> (only events produced after
     * the listener starts). Provide a known event id to resume from it.
     */
    private String lastEventId;

    /**
     * Timeout in milliseconds for each blocking read of the event stream. Default 10000.
     */
    private int blockingTimeout = 10000;

    public boolean isAutorun() {
        return autorun;
    }

    public void setAutorun(boolean autorun) {
        this.autorun = autorun;
    }

    public String getLastEventId() {
        return lastEventId;
    }

    public void setLastEventId(String lastEventId) {
        this.lastEventId = lastEventId;
    }

    public int getBlockingTimeout() {
        return blockingTimeout;
    }

    public void setBlockingTimeout(int blockingTimeout) {
        this.blockingTimeout = blockingTimeout;
    }
}